src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2835,6 +2835,24 @@ LogicalResult ONNXSplitV11Op::inferShapes(
 // Flatten
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXFlattenOp op) {
+
+  auto inTy = op.input().getType().dyn_cast<RankedTensorType>();
+  if (!inTy) {
+    return success();
+  }
+
+  int64_t axisValue = op.axis();
+  auto inputShape = inTy.getShape();
+  int64_t inputRank = inputShape.size();
+
+  if (axisValue < -1 * inputRank || axisValue > inputRank) {
+    return op.emitError("ONNXFlattenOP: axis() value is out of range");
+  }
+
+  return success();
+}
+
 LogicalResult ONNXFlattenOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
   auto inTy = input().getType().dyn_cast<RankedTensorType>();
```

Comment:
Here, auto inTy = input().getType().dyn_cast<RankedTensorType>(); could fail. So you can either use the 'ISA' trick and then you are guaranteed to have this work (when ISA fails, return success)
Or you can use a "dyn_cast_or_null" and test if the result is not null.